### PR TITLE
Adds Redis to rivescript cache, clear cache parameter [pr]

### PR DESCRIPTION
**Pull request by aaronschachter**
#### What's this PR do?

* Refactors the rivescript cache to use Redis, following by @rapala61's example in gambit-campaigns (e.g. https://github.com/DoSomething/gambit-campaigns/pull/995)

* Adds a `cache` query parameter to the `GET /rivescript` endpoint introduced in #393, unblocking adding a cache clear button into Gambit Admin to trigger dynos to stream Rivescript from our cache, and call `sortReplies`
* I've added the Heroku Redis to staging to test this PR; we will need to add to prod if we go this route.

#### How should this be reviewed?
Verify a `GET /api/v2/rivescript?cache=false` request queries the Content API, verify excluding the query parameter does not query the Content API if the bot has already been loaded.

#### Any background context you want to provide?

* Our cache clearing request needs to reload the bot, which is why it makes sense to set the `hasSortedReplies` to false per https://github.com/DoSomething/gambit-conversations/pull/390#pullrequestreview-150584626 -- but I think we may possibly still be in the same boat of each inbound message being retried multiple times until a reply is sent (we've seen a surge of the "Still sorting bot replies, please retry your message" errors since deploying [3.1.0](https://github.com/DoSomething/gambit-conversations/releases/tag/3.1.0) today).

* Similar to #392, this skips out on unit tests at the moment for sake of validating whether this approah will work for us.

#### Relevant tickets
Potentially fixes https://www.pivotaltracker.com/story/show/158037562 and #393 

#### Checklist
- [x] Documentation added for new features/changed endpoints.
- [x] Tests added for new features/bug fixes.
- [x] Tested on staging.

### CODE REVIEW

File: lib/helpers/cache.js [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+const redisClient = require('../../config/redis')();
 
 const config = require('../../config/lib/helpers/cache');
 
-// TODO: Add Redis
-const rivescriptCache = new Cacheman(config.rivescript.name, config.rivescript.ttl);
+const redisEngine = new RedisEngine(redisClient);
+
+const rivescriptCache = new Cacheman(config.rivescript.name, {
+  ttl: config.rivescript.ttl,
```
Comment (rapala61) [inline lib/helpers/cache.js]: We can't ditch the `ttl` in cacheman so we would have to set it to something ridiculous like [100 years](https://github.com/cayasso/cacheman/issues/31#issuecomment-247823203).

File: lib/rivescript.js [hunk 2]
```diff
@@ -14,18 +14,22 @@ function getBot() {
...
-
+/**
+ * @param {Array} rivescripts
+ */
 function streamAndSortRepliesWithRivescripts(rivescripts) {
-  rivescripts.forEach(item => brain.stream(item, logStreamError));
+  rivescripts.forEach(item => brain
+    .stream(item, error => logger.error('error streaming rivescript', { error })));
   logger.info('rivescript.brain.stream success');
   brain.sortReplies();
```
Comment (rapala61) [inline lib/rivescript.js]: Just double checking, is `sortReplies` a promise? since we updated rivescript to 2.x which uses async/await.

Comment (rapala61) [reply lib/rivescript.js]: Great, thanks for checking!

File: lib/helpers/rivescript.js [hunk 3]
```diff
@@ -8,29 +8,35 @@ const config = require('../../config/lib/helpers/rivescript');
...
 
 /**
- * @param {Object} query
  * @return {Promise}
  */
 async function getRivescripts() {
   logger.debug('getRivescripts');
   const cache = await helpers.cache.rivescript.get(cacheKey);
   if (cache) {
+    logger.debug('rivescript cache hit');
```
[the remark this answers is not in the document]
Comment (aaronschachter) [reply lib/helpers/rivescript.js]: Yep -- we have similar debug statements in other catch getters to verify cache hits.

**Comment by aaronschachter** [comment]
Update re: https://github.com/DoSomething/gambit-conversations/pull/394#issuecomment-417203250 is that this seems to be something up with our long code and AT&T, and is unrelated to this branch. See [Slack thread](https://dosomething.slack.com/archives/C2C8NLNAY/p1535674182000100).

**Comment by aaronschachter** [comment]
I've added unit tests and tested the latest branch on staging, looking good. Also added the Heroku Redis add-on to production, so I think this is good to go!

**Pull request merged**